src/main/java/hudson/maven/MavenModuleSetBuild.java

```diff
@@ -1438,6 +1438,23 @@ private String getRootPath(String prefix) {
 
         private static final long serialVersionUID = 1L;
     }
+
+    /**
+     * Check if the current build is a release or not.
+     * If all its maven modules are not SNAPSHOT then it is a release
+     *
+     * @since 2.12.2
+     * @return true if {@link MavenModuleSetBuild} is a release
+     */
+    protected Boolean isRelease() {
+        MavenModuleSet mavenModuleSet = this.getProject();
+        for (MavenModule mavenModule : mavenModuleSet.getModules()) {
+            if(mavenModule.getVersion().contains("-SNAPSHOT")) {
```

Comment:
why not using https://maven.apache.org/ref/3.3.9/maven-artifact/apidocs/org/apache/maven/artifact/ArtifactUtils.html there is a method 	isSnapshot(String version)